app/src/main/java/com/nutomic/syncthingandroid/activities/WebGuiActivity.java

```diff
@@ -171,7 +172,11 @@ public void onServiceStateChange(SyncthingService.State newState) {
             if (mWebView.getUrl() == null) {
                 mWebView.stopLoading();
                 setWebViewProxy(mWebView.getContext().getApplicationContext(), "", 0, "localhost|0.0.0.0|127.*|[::1]");
-                mWebView.loadUrl(getService().getWebGuiUrl().toString());
+                String credentials = mConfig.getUserName() + ":" + mConfig.getApiKey();
+                String b64Credentials = Base64.encodeToString(credentials.getBytes(UTF_8), Base64.NO_WRAP);
```

Comment:
I haven't delved too deeply into how the app handles authentication, but my understanding is that the password and API key are initialized with the same value and can therefore be used interchangeably.

Reply:
If the `X-API-Key` header also works reliably for the UI, we can drop the whole basic auth shenanigans.

Reply:
Ok, that explains why it works. But yeah setting the authorization value to `bearer <api-key>` should work as well and be more straightforward.

Reply:
Not quite, actually :sweat_smile: `Authorization: Bearer` is checked in the [`hasValidAPIKeyHeader`](https://github.com/syncthing/syncthing/blob/5eb20580b19e75d855bebd665c6978a2085025df/lib/api/api_auth.go#L83-L86) code branch, which if successful just allows the request but does _not_ set the session cookie (because it is assumed the client will include the header with each request instead). In order to get a session cookie, you need to log in either via the HTML form or use `Authorization: Basic`, which [does set the session cookie](https://github.com/syncthing/syncthing/blob/5eb20580b19e75d855bebd665c6978a2085025df/lib/api/api_auth.go#L100-L101). And `Authorization: Basic` does not recognize the API key (unless it happens to equal the password).

Reply:
I suppose this means this still doesn't really work, though, because the password is not stored in cleartext in the config, so we can't reliably retrieve it to include in an `Authorization: Basic` header? And `Authorization: Bearer` will work reliably with the API key, but will not set a session cookie and thus won't work past the initial request.

Perhaps `Authorization: Bearer` should set a session cookie, while `X-Api-Key` could continue to not set the cookie. I'm thinking that using the `Authorization` header could indicate an initial authentication request for a session cookie to use for subsequent requests, while using `X-Api-Key` could instead indicate that no cookie is needed because `X-Api-Key` will be included each time instead.

For example: if one is using cURL, it's easier to just keep `-H 'X-Api-Key: <key>` in each command line than having to set a "cookie jar" and such. And the opposite use case is of course this WebView where you can set the headers only for the initial request.

Reply:
Basic authentication works for the app because it configures Syncthing so that the password and the API key matches.

But as @imsodin already pointed out, this is rather an ungly hack.

Could we set the session cookie depending on the context? As this the cookie is rather intended for to the UI, we could exclude `/rest/*` from this behaviour.

Reply:
I wasn't so much saying it's ugly, just that (I thought) it could be done simpler - apparently I didn't grasp the session cookie part. So lets just keep sending basic auth with username/password (which also happens to be the api-key). Sorry for the confusion I caused here.